File: src/frontend/src/admin/types/feedbackReport.ts
export const FeedbackType = {
  BugReport: 'BugReport',
  FeatureRequest: 'FeatureRequest',
  General: 'General',
} as const
export type FeedbackType = (typeof FeedbackType)[keyof typeof FeedbackType]

export const FeedbackStatus = {
  New: 'New',
  InReview: 'InReview',
  Resolved: 'Resolved',
  Closed: 'Closed',
} as const
export type FeedbackStatus = (typeof FeedbackStatus)[keyof typeof FeedbackStatus]

export const FeedbackTypeLabels: Record<FeedbackType, string> = {
  [FeedbackType.BugReport]: 'Bug Report',
  [FeedbackType.FeatureRequest]: 'Feature Request',
  [FeedbackType.General]: 'General',
}

export const FeedbackStatusLabels: Record<FeedbackStatus, string> = {
  [FeedbackStatus.New]: 'New',
  [FeedbackStatus.InReview]: 'In Review',
  [FeedbackStatus.Resolved]: 'Resolved',
  [FeedbackStatus.Closed]: 'Closed',
}

export interface FeedbackReportDto {
  id: string
  referenceNumber: string
  type: FeedbackType
  status: FeedbackStatus
  title: string
  severity: string | null
  contentJson: string | null
  reporterEmail: string
  reporterName: string | null
  userRole: string | null
  orgName: string | null
  orgRole: string | null
  currentUrl: string | null
  screenSize: string | null
  appVersion: string | null
  commitSha: string | null
  exerciseId: string | null
  exerciseName: string | null
  exerciseRole: string | null
  adminNotes: string | null
  gitHubIssueNumber: number | null
  gitHubIssueUrl: string | null
  createdAt: string
  updatedAt: string
}

export interface UpdateFeedbackStatusRequest {
  status: FeedbackStatus
  adminNotes?: string | null
}

export interface PaginationInfo {
  totalCount: number
  totalPages: number
  page: number
  pageSize: number
}

export interface FeedbackListResponse {
  reports: FeedbackReportDto[]
  pagination: PaginationInfo
}

export interface FeedbackQueryParams {
  page?: number
  pageSize?: number
  search?: string
  type?: FeedbackType | null
  status?: FeedbackStatus | null
  sortBy?: string
  sortDesc?: boolean
}

File: src/Cadence.Core/Features/Feedback/Models/Entities/FeedbackReport.cs
using Cadence.Core.Features.Feedback.Models.Enums;
using Cadence.Core.Models.Entities;

namespace Cadence.Core.Features.Feedback.Models.Entities;

/// <summary>
/// Persisted feedback submission. NOT org-scoped — feedback is system-wide
/// so that administrators can view all reports regardless of organization.
/// </summary>
public class FeedbackReport : BaseEntity
{
    /// <summary>Human-readable reference number (e.g. CAD-20260304-1234).</summary>
    public string ReferenceNumber { get; set; } = string.Empty;

    public FeedbackType Type { get; set; }

    public FeedbackStatus Status { get; set; } = FeedbackStatus.New;

    /// <summary>Title (bug/feature) or Subject (general).</summary>
    public string Title { get; set; } = string.Empty;

    /// <summary>Bug severity level. Null for non-bug reports.</summary>
    public string? Severity { get; set; }

    /// <summary>Full request payload stored as JSON for type-specific fields.</summary>
    public string? ContentJson { get; set; }

    // ── Reporter identity (from JWT claims) ──

    public string ReporterEmail { get; set; } = string.Empty;
    public string? ReporterName { get; set; }
    public string? UserRole { get; set; }
    public string? OrgName { get; set; }
    public string? OrgRole { get; set; }

    // ── Client context (from FeedbackClientContext) ──

    public string? CurrentUrl { get; set; }
    public string? ScreenSize { get; set; }
    public string? AppVersion { get; set; }
    public string? CommitSha { get; set; }
    public string? ExerciseId { get; set; }
    public string? ExerciseName { get; set; }
    public string? ExerciseRole { get; set; }

    // ── Admin triage ──

    /// <summary>Admin notes added during triage/review.</summary>
    public string? AdminNotes { get; set; }

    // ── GitHub integration ──

    /// <summary>GitHub issue number created for this report. Null if not configured or creation failed.</summary>
    public int? GitHubIssueNumber { get; set; }

    /// <summary>Full URL to the GitHub issue.</summary>
    public string? GitHubIssueUrl { get; set; }
}
